website/Makefile

```diff
@@ -1,58 +1,57 @@
+
+this_dir := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+chisel_dir := $(realpath $(this_dir)/..)
+relative_to_this_dir := $(notdir $(this_dir))
+
+mdoc_out := $(chisel_dir)/docs/generated
+
 buildDir ?= build
 subprojects = $(buildDir)/subprojects
 docs = $(buildDir)/docs
 website_docs_dest = docs/src/main/tut/chisel3/docs
 chisel3 = ../
+install_timestamp = $(this_dir)/node_modules/timestamp
 
-www-docs = \
-	$(shell find $(chisel3)/docs/ -name "*.md")
+# Everything not in the website directory
+# This is overly conservative but okay until we can do something more precise
+chisel-non-website = \
+  $(shell git ls-files $(chisel_dir) --exclude-per-directory $(relative_to_this_dir))
 
-www-src = \
-	$(website_docs_dest) \
-	$(shell find docs/src/main/resources) \
-	$(shell find docs/src/main/tut) \
-	$(chisel3)/README.md
+default: build
 
-# Get the latest version of some sequence of version strings
-# Usage: $(call getTags,$(foo))
-define latest
-$(shell echo $(1) | tr " " "\n" | sort -Vr | head -n1 | sed 's/^v//')
-endef
+# Using introduction as standin for all md files
+# Note this doesn't work when a markdown file is deleted
+$(mdoc_out)/introduction.md: $(chisel-non-website)
+	cd $(chisel_dir) && sbt docs/mdoc
 
+$(this_dir)/docs/introduction.md: $(mdoc_out)/introduction.md
+	mkdir -p $(shell dirname $@)
+	cp -R $(mdoc_out)/* $(this_dir)/docs/
 
-.PHONY: all clean mrproper publish serve
-.PRECIOUS: \
-	$(subprojects)/chisel3/%/.git $(subprojects)/chisel3/%/target/scala-$(scalaVersion)/unidoc/index.html
+contributors = $(this_dir)/src/pages/generated/contributors.md
 
-# Build the site into the default directory (docs/target/site)
-all: docs/target/site/index.html
+# Unclear what this should depend on
+$(contributors): $(chisel-non-website)
+	mkdir -p $(shell dirname $@)
```

Comment:
See above comment about order-only prerequisites. 

Unrelated if the above comment is applied: `$(shell dirname $@)` is equivalent to `$(@D)`. All the uncommon, but useful automatic variables are listed out here: https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html